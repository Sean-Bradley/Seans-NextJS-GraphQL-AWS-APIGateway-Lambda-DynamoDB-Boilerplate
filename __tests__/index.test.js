/* eslint-env jest */

import { shallow } from 'enzyme'
import React from 'react'
import renderer from 'react-test-renderer'

import App from '../pages/index.js'

describe('With Enzyme', () => {
  it('App shows "seans nextjs, graphql, aws, apigateway, lambda, dynamodb boilerplate"', () => {
    const app = shallow(<App />)
    expect(app.find('h2').text()).toEqual('seans nextjs, graphql, aws, apigateway, lambda, dynamodb boilerplate')
  })
})

describe('With Snapshot Testing', () => {
  it('App shows "seans nextjs, graphql, aws, apigateway, lambda, dynamodb boilerplate"', () => {
    const component = renderer.create(<App />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
