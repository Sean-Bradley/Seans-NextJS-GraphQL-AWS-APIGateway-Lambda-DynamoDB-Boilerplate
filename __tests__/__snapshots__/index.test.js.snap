// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing App shows "seans nextjs, graphql, aws, apigateway, lambda, dynamodb boilerplate" 1`] = `
<div
  className="container"
>
  <div
    className="jumbotron text-center"
  >
    <h2>
      seans nextjs, graphql, aws, apigateway, lambda, dynamodb boilerplate
    </h2>
    <p>
      Boilerplate for NextJS, which uses ReactJS and NodeJS, sending GraphQL to an AWS Api Getway, which fronts an AWS Lambda, which Fronts an AWS DynamodDB.
    </p>
  </div>
  <p>
    Nothing to see here, go and see 
    <a
      href="/cats"
      onClick={[Function]}
    >
      Cats
    </a>
     instead.
  </p>
</div>
`;
